### ISSUE TITLE
[Feature]: Increase the JSON forms field limit

### ISSUE THREAD

**Issue by marcjaner** [appsmithorg/appsmith#35177]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

There is a 50 fields limit in JSON forms, which becomes a huge limitation when building pages for complex use cases as more fields might be needed in a JSON form. This is a long known [issue](https://github.com/appsmithorg/appsmith/issues/13165) and a lot of people are complaining because of it.

### Why should this be worked on?

Apparently, changing this should not be too complex, as it is a constant defined in [here](https://github.com/appsmithorg/appsmith/blob/18a4b5046b0feafd9fc312585dd5356271865796/app/client/src/widgets/JSONFormWidget/constants.ts#L171) as:
`export const MAX_ALLOWED_FIELDS = 50;`

**Comment by nirkons** [comment]
> @marcjaner unfotunately we haven;t been able to get to this issue and since it's already documented, I'm going to close this issue. The challenge is not about increasing the limit but about the performance of the form once the limit is increased. In general forms of 50+ fields are a bad experience for end users and don't make sense.

regarding the logic behind this - it "doesn't make sense" for you since you are thinking about this form like it's a google form for example, but I would wager that is **not** the main use case for most people. For me (and others), its just a container of fields and some buttons, not all fields have to be filled in at a time, data is extracted, and the user updates only what they want. Instead of this "form", if there was an easy way to create 100 fields from a JSON payload, have it mapped automatically (including types) and it was not called a "form" then it would resolve the need.

Basically what I'm saying is the reluctance to increase the maximum doesn't actually help anything either way (even in terms of performance), because I needed to create a container with about 100 fields, so it only meant that I needed to add 100 fields and map them individually (taking hours) instead of minutes, would the performance be different if each individual field was mapped or it was used in a JSON form?
